include/fluent-bit/flb_log_event_decoder.h

```diff
@@ -56,6 +58,7 @@ struct flb_log_event_decoder {
     const char       *buffer;
     size_t            offset;
     size_t            length;
+    int               decode_result;
```

Comment:
I added a decoding result as a member to prevent passing it as an arg.

I don't want to do like
```c
        while ((result =flb_log_event_decoder_next(
                                           &log_decoder,
                                           &log_event)) == FLB_EVENT_DECODER_SUCCESS) {
               ...
       }

       result = flb_log_event_decoder_get_decode_result(result, &log_decoder);
```

Reply:
Rename the field to `last_result`, this field could be set while initializing the decoder so that could be confusing.